### PR TITLE
INTMDB-602: database user profile changes

### PR DESCRIPTION
**Pull request by govardhanpagidi**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: cfn-resources/database-user/resource-role.yaml [hunk 1]
```diff
@@ -31,6 +31,20 @@ Resources:
                 - "secretsmanager:UpdateSecretVersionStage"
                 - "ec2:CreateVpcEndpoint"
                 - "ec2:DeleteVpcEndpoints"
+                - "cloudformation:CreateResource"
+                - "cloudformation:DeleteResource"
+                - "cloudformation:GetResource"
```
Comment (andreaangiolillo) [inline cfn-resources/database-user/resource-role.yaml:36]: Why are these needed?

File: cfn-resources/database-user/test/inputs_1_update.template.json [hunk 2]
```diff
@@ -2,11 +2,8 @@
     "Username" : "DataUser1",
     "Password" : "MongoDB12345%",
     "ProjectId" : "ProjectId",
-    "ApiKeys" : {
-        "PublicKey" : "$ATLAS_PUBLIC_KEY",
-        "PrivateKey" : "$ATLAS_PRIVATE_KEY"
-    },
-    "DatabaseName": "admin",
+  "Profile" : "default",
+  "DatabaseName": "admin",
```
Comment (andreaangiolillo) [inline cfn-resources/database-user/test/inputs_1_update.template.json:6]:
```suggestion
     "Profile" : "default",
     "DatabaseName": "admin",
```

**Review by andreaangiolillo — APPROVED** [review]
LGTM. Comments are not blocking and can be addressed in followup PR

File: cfn-resources/database-user/resource-role.yaml (hunk 1 above)
Comment (govardhanpagidi) [reply cfn-resources/database-user/resource-role.yaml:36]: These permissions are generic and the same will applied when we publsih, doesn't harm anything for private extension
